### PR TITLE
tests: Replaces images used with agnhost (part 1)

### PR DESCRIPTION
**Pull request by claudiubelu**
**What type of PR is this?**

/kind feature

/sig testing

**What this PR does / why we need it**:

Quite a few images are only used a few times in a few tests. Thus, the images are being centralized into the agnhost image, reducing the number of images that have to be pulled and used.

This PR replaces the usage of the following images with agnhost:

- fakegitserver
- hostexec
- liveness
- logs-generator
- no-snat-test
- no-snat-test-proxy
- port-forward-tester

**Which issue(s) this PR fixes**:

Related: #76342
Depends On: #78389

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by mkumatag** [comment]
This is happening because of version is not `2.0` in manifest.go file, please change it to `2.0`

error:
```
I0530 10:36:48.507] W0530 10:36:39.838874    2474 image_list.go:160] Failed to pull gcr.io/kubernetes-e2e-test-images/agnhost:1.1 as user "root", retrying in 1s (1 of 5): exit status 1
I0530 10:36:48.507] W0530 10:36:41.028774    2474 image_list.go:160] Failed to pull gcr.io/kubernetes-e2e-test-images/agnhost:1.1 as user "root", retrying in 1s (2 of 5): exit status 1
I0530 10:36:48.507] W0530 10:36:42.225168    2474 image_list.go:160] Failed to pull gcr.io/kubernetes-e2e-test-images/agnhost:1.1 as user "root", retrying in 1s (3 of 5): exit status 1
I0530 10:36:48.508] W0530 10:36:43.430250    2474 image_list.go:160] Failed to pull gcr.io/kubernetes-e2e-test-images/agnhost:1.1 as user "root", retrying in 1s (4 of 5): exit status 1
I0530 10:36:48.508] W0530 10:36:44.641279    2474 image_list.go:160] Failed to pull gcr.io/kubernetes-e2e-test-images/agnhost:1.1 as user "root", retrying in 1s (5 of 5): exit status 1
I0530 10:36:48.508] W0530 10:36:44.641331    2474 image_list.go:164] Could not pre-pull image gcr.io/kubernetes-e2e-test-images/agnhost:1.1 exit status 1 output: Pulling repository gcr.io/kubernetes-e2e-test-images/agnhost
I0530 10:36:48.509] unauthorized: authentication required
```

**Comment by bclau** [comment]
> 
> 
> This is happening because of version is not `2.0` in manifest.go file, please change it to `2.0`
> 

Oups, I forgot to update this PR.

**Comment by bclau** [comment]
Well, at least this passed.

File: test/e2e/network/no_snat.go [hunk 1]
```diff
@@ -40,8 +40,8 @@ const (
 	testProxyPort = 31235 // Firewall rule allows external traffic on ports 30000-32767. I just picked a random one.
 )
 
-var testPodImage = imageutils.GetE2EImage(imageutils.NoSnatTest)
-var testProxyImage = imageutils.GetE2EImage(imageutils.NoSnatTestProxy)
+var testPodImage = imageutils.GetE2EImage(imageutils.Agnhost)
+var testProxyImage = imageutils.GetE2EImage(imageutils.Agnhost)
```
Comment (spiffxp) [inline test/e2e/network/no_snat.go]: nit: these don't seem to be re-used elsewhere, I find it more legible to have the `imageutils` decl right next to the args being used

**Review by spiffxp — COMMENTED** [review]
/approve
non-blocking nit but otherwise looks good (the hostexec container rename sent me spelunking for a bit to see if anything looked for containers by name, but couldn't find anything)

**Comment by spiffxp** [comment]
/hold
/priority important-soon
pending merge of #78389

File: test/e2e/auth/node_authn.go [hunk 2]
```diff
@@ -97,7 +97,7 @@ func createNodeAuthTestPod(f *framework.Framework) *v1.Pod {
 		Spec: v1.PodSpec{
 			Containers: []v1.Container{{
 				Name:    "test-node-authn",
-				Image:   imageutils.GetE2EImage(imageutils.Hostexec),
+				Image:   imageutils.GetE2EImage(imageutils.Agnhost),
```
Comment (johnSchnake) [inline test/e2e/auth/node_authn.go]: non-blocking nit: I know the default for the Agnhost was to run `pause` but do you think we should make that explicit via the args here?

Comment (bclau) [reply test/e2e/auth/node_authn.go]: we could. It's probably safer. Then we'll have to do the same everywhere.

File: test/e2e/network/no_snat.go (hunk 1 above)
Comment (bclau) [reply test/e2e/network/no_snat.go]: makes sense.

**Comment by spiffxp** [comment]
/lgtm
(still waiting on https://github.com/kubernetes/kubernetes/pull/78389 to merge)

File: test/utils/image/manifest.go [hunk 3]
```diff
@@ -232,7 +218,7 @@ func initImageConfigs() map[int]Config {
 	configs := map[int]Config{}
 	configs[CRDConversionWebhook] = Config{e2eRegistry, "crd-conversion-webhook", "1.13rev2"}
 	configs[AdmissionWebhook] = Config{e2eRegistry, "webhook", "1.15v1"}
-	configs[Agnhost] = Config{e2eRegistry, "agnhost", "1.0"}
+	configs[Agnhost] = Config{e2eRegistry, "agnhost", "2.0"}
```
Comment (spiffxp) [inline test/utils/image/manifest.go]: should this be `2.1` or does this need to be done as a followup?

**Comment by spiffxp** [comment]
/lgtm cancel
actually, versioning question

File: test/utils/image/manifest.go (hunk 3 above)
Comment (bclau) [reply test/utils/image/manifest.go]: 2.0.

It s set to 2.1 in part 2: https://github.com/kubernetes/kubernetes/pull/78396/files#diff-d511d08b2cdc3d78d9e75c3c131ed2ceR226

**Comment by bclau** [comment]
It seems that this PR is being held. Any reason why it is?

**Comment by spiffxp** [comment]
/hold cancel
@bclau description said this PR depended on https://github.com/kubernetes/kubernetes/pull/78389 so I applied a hold saying we'll wait for it to merge (which it now has)

**Pull request merged**